lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1244,16 +1244,21 @@ def has_dict_changed(new_dict, old_dict):
     return False
 
 
-def has_list_changed(new_list, old_list):
+def has_list_changed(new_list, old_list, sort_lists=True):
     """
-    Check two lists has differences.
+    Check two lists have differences. Sort lists by default.
     """
     if new_list is None:
         return False
     old_list = old_list or []
     if len(new_list) != len(old_list):
         return True
-    for new_item, old_item in zip(new_list, old_list):
+
+    if sort_lists:
+        zip_data = zip(sort(new_list), sort(old_list))
```

Comment:
```suggestion
        zip_data = zip(sorted(new_list), sorted(old_list))
```

Reply:
Also note that simply sorting does not necessarily give the correct result for lists of dicts. So this change could actually recreate/update services where nothing changed.

Reply:
Fixed using sorted over sort. You're right about the list of dictionaries issue, and I don't have a good solution here. This should be an improvement over what's currently being performed.